### PR TITLE
[WNMGDS-1610] Button analytics

### PR DESCRIPTION
**Pull request by pwolfert**
## Summary

https://jira.cms.gov/browse/WNMGDS-1610

### Added

- Added analytics tracking for `Button` element. Can be turned on by calling `setButtonSendsAnalytics(true)`.

## How to test

1. `yarn storybook`
2. Navigate to http://localhost:6006/?path=/story/components-button--default-button&globals=analytics:on
3. Click on the button and make sure the event is firing as expected in the console
4. Navigate to http://localhost:6006/?path=/story/components-button--link-button&globals=analytics:on
5. Click on the button and make sure the event is firing as expected in the console

### CODE REVIEW

**Comment by pwolfert** [comment]
In order to test this easily, https://github.com/CMSgov/design-system/pull/1783 should be merged

**Comment by zarahzachz** [comment]
I'm having trouble understanding what results I'm expected to see in the console. 

In the browser console (FF and Chrome), I see nothing returned.

In the Actions tab, I see a click event firing, but no analytics results returned (unless they're nested within something?).
Screen Shot 2022-05-03 at 10 36 55 AM

**Comment by forrestbaer** [comment]
It's working for me when I enable the `BUTTON_SENDS_ANALYTICS` flag but the storybook controls and args seem out of sync, only the boolean toggle in the attributes panel seems to make any difference to whether the analytics are shown or not. The on/off toggle with the attributes panel are out of sync and query strings like `&args=analytics:false&globals=analytics:on` are being built with some clicking around.

**Comment by forrestbaer** [comment]
Looks like 4 tests fail when the `BUTTON_SENDS_ANALYTICS` flag is set to true.

**Comment by pwolfert** [comment]
@zarahzachz, it turns out I actually never updated that Storybook tool after merging that branch. I needed to have it turn on button analytics too.

File: packages/design-system/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -144,20 +150,68 @@ export const Button = ({
...
-
   if (ComponentType !== 'button') {
     // Assume `component` is not a custom component rendering a <button>
     // and remove <button> specific attributes
     delete attrs.disabled;
     delete attrs.type;
   }
 
+  function sendButtonEvent() {
+    if (!buttonSendsAnalytics() || analytics === false) {
```
[the remark this answers is not in the document]
Comment (pwolfert) [reply packages/design-system/src/components/Button/Button.tsx:161]: This is set up with a purpose; it allows teams to opt-in to component analytics overall but opt-out of it for a particular instance of that component. One example of that early on was to be able to disable it for a paricular component that contained PII.

[the remark this answers is not in the document]
Comment (pwolfert) [reply packages/design-system/src/components/Button/Button.tsx:161]: There's a good feature idea for next PI: A copy-to-clipboard button

**Review by zarahzachz — APPROVED** [review]
works

**Pull request merged**